167190491 User Login

JuwonAbiola opened the pull request:
#### What does this PR do?
- Enable user login functionality on the Authors Haven app

#### Description of Task proposed in this pull request?
- Build and style the login page
- Consume the login endpoint
- Write component and e2e test

#### How should this be manually tested (Quality Assurance)?
- Run `npm run start:dev'
- A browser tab will launch then navigate to /login
- Input a valid email or username and password

#### What are the relevant pivotal tracker stories?
- [167190491](https://www.pivotaltracker.com/story/show/167190491)

#### Any background context you want to add (Operations Impact)?
- Run npm install to download the project's dependencies

#### What I have learned working on this feature: 
- I learned how to map state to props

#### Screenshots: 
- ![Uploading image.png…]()

dinobi commented on src/styles/index.scss:
```diff
@@ -140,3 +147,30 @@ input[type='password'] {
+    .has-validation-error {
+        border-color:$redColor !important;
+    }
+    
```
add new line

dinobi commented on src/reducers/user/login.js:
```diff
@@ -0,0 +1,39 @@
+  switch (action.type) {
+    case actionTypes.LOGIN_REQUEST:
+      return {
+        state, loggingIn: true, isLoading: true
```
what is `loggingIn`?

dinobi commented on src/reducers/user/login.js:
```diff
@@ -0,0 +1,39 @@
+        isLoading: false
+      };
+    case actionTypes.LOGIN_FAILURE:
+      return {
```
why don't you maintain consistency with your object structure?

abdulfatai360 commented on src/components/Login/login.scss:
```diff
@@ -0,0 +1,104 @@
+}
+.input-validation-error {
+    font-size: 0.95em;
+    color:red !important;
```
@jhayX You didn't remember to replace the `red` value with the `$redColor` variable you added in the `variables.scss` file below.

abdulfatai360 commented on src/components/Login/login.jsx:
```diff
@@ -0,0 +1,138 @@
+                 <img className="google-icon" src={GoogleIcon} alt="google" />
+                 <img className="twitter-icon" src={TwitterIcon} alt="twitter" />
+               </span>
+               <span>Don't have an account? login</span>
```
@jhayX I believe we expect this link should redirect users to the signup page, right? You could make use of the `Link` component of `react-router-dom` package to achieve this.

ayodejiAA replied:
@jhayX In addition, Why should you redirect a user to login form again? Please, let's change this to signup.

abdulfatai360 commented on src/components/Login/login.jsx:
```diff
@@ -0,0 +1,138 @@
+                   disabled={isLoading}
+                 />
+               </div>
+               <p className="center">Forgot Password?</p>
```
@jhayX Could you also kindly redirect users to the `request password` page when they click on this? Kindly take advantage of `Link` component of `react-router-dom` to achieve this

dinobi commented on src/components/Login/login.spec.js:
```diff
@@ -1,12 +1,92 @@
+describe('Login Component interaction', () => {
+  it('updates its state when input fields change', () => {
+    const { wrapper } = setup();
+    const classInstance = wrapper.instance();
```
please use the proper variable name here

dinobi commented on src/components/Login/login.jsx:
```diff
@@ -0,0 +1,144 @@
+    }
+  };
+
+  handleChange=(e) => {
```
put spaces between operators

JuwonAbiola committed:
ft-[user login]
- create and design login page
- consume login endpoint
- write component and e2e test
[Delivers #167190491]
